wrapper-generator/src/main/kotlin/it/krzeminski/githubactions/wrappergenerator/versions/Http.kt

```diff
@@ -0,0 +1,39 @@
+package it.krzeminski.githubactions.wrappergenerator.versions
+
+import it.krzeminski.githubactions.wrappergenerator.domain.ActionCoords
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.builtins.ListSerializer
+import kotlinx.serialization.json.Json
+import okhttp3.OkHttpClient
+import okhttp3.Request
+
+val ActionCoords.apiTagsUrl: String
+    get() = "https://api.github.com/repos/$owner/$name/git/matching-refs/tags/v"
+
+@Serializable
+data class GithubTag(
+    val ref: String,
+)
+
+val json = Json { ignoreUnknownKeys = true }
+
+val okhttpClient by lazy {
+    OkHttpClient()
+}
+
+fun ActionCoords.fetchAvailableVersions(githubToken: String): List<Version> {
+    val request: Request = Request.Builder()
+        .header("Authorization", "token $githubToken")
+        .url(apiTagsUrl)
+        .build()
+
+    val content = okhttpClient.newCall(request).execute().use { response ->
+        if (response.isSuccessful.not()) {
+            println(response.headers)
+            error("API rate reached?  See https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting")
+        }
+        response.body!!.string()
+    }
+    val data = json.decodeFromString(ListSerializer(GithubTag.serializer()), content)
```

Comment:
I think this simpler form will work as well:
```suggestion
    val data = json.decodeFromString<List<GithubTag>>(content)
```

Reply:
No, that doesn't work because of type erasure.

Reply:
"Works on my machine" ;) This overload of `decodeFromString` is `reified`:

```kotlin
fun main() {
    val content = "[{\"ref\": \"some-ref\"}, {\"ref\": \"12334\"}]"
    val data = json.decodeFromString<List<GithubTag>>(content)
    println(data)  // [GithubTag(ref=some-ref), GithubTag(ref=12334)]
}
```

Reply:
You were right, fixed